src/goto-programs/remove_returns.cpp

```diff
@@ -84,6 +84,7 @@ remove_returnst::get_or_create_return_value_symbol(const irep_idt &function_id)
   // Return-value symbols will always be written before they are read, so there
   // is no need for __CPROVER_initialize to do anything:
   new_symbol.type.set(ID_C_no_initialization_required, true);
+  new_symbol.type.set(ID_C_no_nondet_initialization, true);
```

Comment:
Hmm, shouldn't the code instead be looking at the `ID_C_no_initialization_required` flag? It seems we now need to keep to flags in sync, in multiple places? Also, what is wrong with non-det initialising them?

Reply:
They are guaranteed to be assigned. Nondet-initialisation can become very costly for return values that are pointers to general linked data structures.

Reply:
@svorenova, why does `ID_C_no_initialization_required` not imply `ID_C_no_nondet_initialization` ?

Reply:
It appears that only return values have the `ID_C_no_initialization_required` flag. So I can make `nondet-static` ignore all static variables with either of the two flags. Makes sense, thank you!

Reply:
> They are guaranteed to be assigned. Nondet-initialisation can become very costly for return values that are pointers to general linked data structures.

Please capture this information in the commit message.

Reply:
Done